### ISSUE TITLE
grpc-server do not return error when creating bad template

### ISSUE THREAD

**Issue by invidian** [tinkerbell/tink#317]
## Expected Behaviour

If one provides bad template to gRPC call to create a template, error should be returned.

## Current Behaviour

Currently no error is returned, but no template is created either.

## Steps to Reproduce (for bugs)
Add following unit test case:
```patch
diff --git grpc-server/template_test.go grpc-server/template_test.go
index 84fc6b3..fc67766 100644
--- grpc-server/template_test.go
+++ grpc-server/template_test.go
@@ -90,6 +90,18 @@ func TestCreateTemplate(t *testing.T) {
                                expectedError: true,
                        },
                },
+               "BadYAML": {
+                       args: args{
+                               db: mock.DB{
+                                       TemplateDB: make(map[string]interface{}),
+                               },
+                               name:     "template_1",
+                               template: `"foo`,
+                       },
+                       want: want{
+                               expectedError: true,
+                       },
+               },
        }

        for name := range testCases {
```

It can also be reproduced with `tink` CLI:
1. Create file `bar.yaml` with content: `"foo` (intentional not closed quote)
2. Run `template create --name foo --path bar.yaml`
3. Error should be returned.

**Comment by JayHelton** [comment]
Hello 👋 

Hacktober participant here. Tinkerbell is very interesting, and hacktoberfest is the perfect time for me to finally learn and contribute.

I would love to grab this good first issue

**Comment by gianarb** [comment]
Hello @JayHelton ! That's great! please follow one of the setup guides in our documentation https://tinkerbell.org/docs/ and feel free to join our Slack community if you need any help or you have questions https://tinkerbell.org/community/

have a nice hacktoberfest and a good weekend!

**Comment by invidian** [comment]
@JayHelton I think you don't need to set up Tinkerbell to resolve this issue, just in case, as the issue can be reproduced with unit tests.

However, I noticed, that the database mock has different implementation than the actual server, which might make a difference here, as right now the database implementation does the validation of the data. I'm pretty sure validation should be happening before data is written to the database, but it might be out of scope for this issue.

**Comment by mmlb** [comment]
Hey @invidian this is only an issue in the unit tests right?

**Comment by invidian** [comment]
>Hey @invidian this is only an issue in the unit tests right?

No, as mentioned at the bottom of the issue, this can also be reproduced on the running setup. I stumbled upon it while implementing acceptance tests for https://github.com/kinvolk/terraform-provider-tinkerbell.

**Comment by mmlb** [comment]
oh I see I very much missed that. Yes the validation wasn't very strict and #315 should definitely address this.

**Comment by invidian** [comment]
Hm, this actually seems like a bug, as the code for validation is there (https://github.com/tinkerbell/tink/blob/master/db/template.go#L22-L29), but the error seems to be lost somewhere. I checked `db` package unit tests and they fail properly with bad input. But not `grpc-server` package tests.

**Comment by JayHelton** [comment]
Just let me know if this is actually not an issue!

**Comment by mmlb** [comment]
@invidian can you reproduce off of my PR?

**Comment by mmlb** [comment]
@invidian I'm getting:
```
/ # tink template create --name foo --path bar.yaml
2020/10/02 18:41:47 rpc error: code = Unknown desc = parsing yaml data: yaml: line 2: found unexpected end of stream
``` which looks like a straight up YAML error, so not even getting into my PR territory.

**Comment by invidian** [comment]
>@invidian can you reproduce off of my PR?

Just tested with #315 and it seems it behaves correctly. I was previously testing using `quay.io/tinkerbell/tink:sha-adb49da`, which is a default image in https://github.com/tinkerbell/sandbox.

**Comment by mmlb** [comment]
oic. Yeah sandbox doesn't use master (on purpose) and I don't think its been updated to post template validation. I think we should update sandbox after my PR is merged @gianarb.

**Comment by mmlb** [comment]
Closing as not a bug then

**Comment by invidian** [comment]
Thanks and sorry I didn't check how it behaves on master branch. Bisected it so it has indeed been fixed in 845cc32b377e730c4f8b9891450444bab003fef5, cool!

**Comment by mmlb** [comment]
:+1: np at all.
